test/functional/autocmd/textyankpost_spec.lua

```diff
@@ -0,0 +1,166 @@
+local helpers = require('test.functional.helpers')
+local clear, eval, eq, insert = helpers.clear, helpers.eval, helpers.eq, helpers.insert
+local feed, execute, expect = helpers.feed, helpers.execute, helpers.expect
+
+
+describe('TextYankPost', function()
+  before_each(function()
+    clear()
+
+    -- emulate the clipboard so system clipboard isn't affected
+    execute('let &rtp = "test/functional/fixtures,".&rtp')
+
+    execute('let g:count = 0')
+    execute('autocmd TextYankPost * let g:event = copy(v:event)')
```

Comment:
With the current code you do not need `copy()` here. It is better to test without `copy()` because I am pretty sure plugin writers will not use it until you start reusing one and the same dictionary in place of always setting `v:event` to new one.